resources/views/community/components/user/profile/primary-meta.blade.php

```diff
@@ -0,0 +1,88 @@
+@props([
+    'hardcoreRankMeta' => [],
+    'softcoreRankMeta' => [],
+    'userMassData' => [],
+    'username' => '',
+])
+
+<?php
+use App\Site\Enums\Permissions;
+use Illuminate\Support\Carbon;
+
+$hasVisibleRole = $userMassData['Permissions'] !== Permissions::Registered && $userMassData['Permissions'] !== Permissions::Unregistered;
```

Comment:
I wonder if we shouldn't show Unregistered (at least for admins). When a new user complains about not being able to unlock stuff, it's often because they haven't verified their email yet.

Admittedly, that information is already available in the Admin panel, but a big tag would make it obvious.

Reply:
Spam, Banned, Unregistered, and Registered should completely be stripped from presentation on public profile information. Those are all accessible in the admin panels and considered relevant only for moderation imo.

Spam and Banned will result in a 404, hidden from the public eye.
Registered and Unregistered are just distinctive to email verification which is a moderation issue and should be considered private information. not having a verified email address should inform the features a user can use. Personally, I'd restrict that heavily. Anyways, non of the public's business imo.

Reply:
> Admittedly, that information is already available in the Admin panel, but a big tag would make it obvious.

We can think about adding a moderation card on profiles to have quick access to a user's activities and status, but should follow the introduction of the admin panel to make sure that's the primary interface to interact with a user on a moderation level.

Reply:
Things will probably change as we move to Roles, but the Spam/Banned/Unregistered information that is currently available is being eliminated by this PR:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/74a603ec-6976-45c5-8e90-4b6d84343d66)

Spam and Banned do result in 404 for non-Admins, so having those accounts show the Spam/Banned tag is useful information - an Admin visiting a profile may not realize the account has been moderated.

I agree that the average user doesn't need to know the difference between whether or not another user is Registered or Unregistered, but it is useful information for diagnosing a specific common issue for new users. 

As I mentioned earlier, the user's Permissions setting is available in the Admin panel, as Moderators are allowed to change the value from there. It's just that the previously visible information is now hidden behind a page scroll and a click.

Reply:
Got it, that makes perfectly sense now. I interpreted that incorrectly :)

Reply:
Logic adjusted in latest to show all `Permissions` strings for admins except "Registered".